### PR TITLE
t8_test_shmem to t8_gtest_shmem

### PR DESCRIPTION
**Pull request by WeberJulia**
**_Describe your changes here:_**

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

- [ ] The author added a BSD statement to `doc/` (or already has one)
- [ ] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

- [ ] All tests pass (in various configurations, this should be executed automatically in a github action)
- [ ] New source/header files are properly added to the Makefiles
- [ ] New Datatypes are added to t8indent_custom_datatypes.txt
- [ ] The reviewer executed the new code features at least once and checked the results manually
- [ ] The code is covered in an existing or new test case
- [ ] New tests use the Google Test framework
- [ ] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [ ] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)
- [ ] Testing of this template: If you feel something is missing from this list, contact the developers

### CODE REVIEW

File: test/t8_data/t8_gtest_shmem.cxx [hunk 1]
```diff
@@ -20,32 +20,50 @@
...
+  sc_mpi_comm_get_node_comms (comm, &intranode, &internode);
 
 #if T8_ENABLE_MPI
   /* Check that they are not NULL */
-  SC_CHECK_ABORT (intranode != sc_MPI_COMM_NULL,
-                  "intra node communicator not set.");
-  SC_CHECK_ABORT (internode != sc_MPI_COMM_NULL,
-                  "inter node communicator not set.");
+  EXPECT_TRUE(intranode != sc_MPI_COMM_NULL);
+  EXPECT_TRUE(internode != sc_MPI_COMM_NULL);
```
Comment (jmark) [inline test/t8_data/t8_gtest_shmem.cxx:63]: Please transfer custom error message.

File: test/t8_data/t8_gtest_shmem.cxx [hunk 2]
```diff
@@ -55,28 +73,94 @@ t8_test_shmem_init_finalize (sc_MPI_Comm mpic)
...
 
   /* Get intranode and internode comm */
-  sc_mpi_comm_get_node_comms (mpic, &intranode, &internode);
+  sc_mpi_comm_get_node_comms (comm, &intranode, &internode);
   /* Check that they are NULL */
-  SC_CHECK_ABORT (intranode == sc_MPI_COMM_NULL,
-                  "intra node communicator not set.");
-  SC_CHECK_ABORT (internode == sc_MPI_COMM_NULL,
-                  "inter node communicator not set.");
+  EXPECT_TRUE(intranode == sc_MPI_COMM_NULL);
```
Comment (jmark) [inline test/t8_data/t8_gtest_shmem.cxx:82]: Please transfer custom error message.

File: test/t8_data/t8_gtest_shmem.cxx [hunk 3]
```diff
@@ -55,28 +73,94 @@ t8_test_shmem_init_finalize (sc_MPI_Comm mpic)
...
+  EXPECT_TRUE(internode == sc_MPI_COMM_NULL);
```
Comment (jmark) [inline test/t8_data/t8_gtest_shmem.cxx:83]: Please transfer custom error message.

File: test/t8_data/t8_gtest_shmem.cxx [hunk 4]
```diff
@@ -90,39 +174,37 @@ t8_test_shmem_array (sc_MPI_Comm comm)
...
+    
     /* setup shared memory usage */
     t8_shmem_init (comm);
     t8_shmem_set_type (comm, shmem_type);
 
 #if T8_ENABLE_MPI
     const sc_shmem_type_t control_shmem_type = sc_shmem_get_type (comm);
-    SC_CHECK_ABORT (shmem_type == control_shmem_type,
-                    "Setting shmem type not succesful.\n");
+    EXPECT_EQ(shmem_type, control_shmem_type);
```
Comment (jmark) [inline test/t8_data/t8_gtest_shmem.cxx:188]: Please transfer custom error message.

File: test/t8_data/t8_gtest_shmem.cxx [hunk 5]
```diff
@@ -90,39 +174,37 @@ t8_test_shmem_array (sc_MPI_Comm comm)
...
 
     /* Allocate one integer */
     t8_shmem_array_t    shmem_array;
     t8_shmem_array_init (&shmem_array, element_size, array_length, comm);
 
     sc_MPI_Comm         check_comm = t8_shmem_array_get_comm (shmem_array);
-    SC_CHECK_ABORT (comm == check_comm,
-                    "shared memory array has wrong communicator.\n");
+    /* Check communicator of shared memory array. */
+    EXPECT_EQ(comm, check_comm);
```
Comment (jmark) [inline test/t8_data/t8_gtest_shmem.cxx:197]: Please transfer custom error message.

File: test/t8_data/t8_gtest_shmem.cxx [hunk 6]
```diff
@@ -90,39 +174,37 @@ t8_test_shmem_array (sc_MPI_Comm comm)
...
 
+    /* Check element count of shared memory array. */
     const size_t        check_count =
       t8_shmem_array_get_elem_count (shmem_array);
-    SC_CHECK_ABORT (check_count == array_length,
-                    "shared memory array has wrong element count.\n");
+    EXPECT_TRUE(check_count == array_length);
```
Comment (jmark) [inline test/t8_data/t8_gtest_shmem.cxx:202]: Please transfer custom error message.

File: test/t8_data/t8_gtest_shmem.cxx [hunk 7]
```diff
@@ -90,39 +174,37 @@ t8_test_shmem_array (sc_MPI_Comm comm)
...
 
+    /* Check element size of shared memory array. */
     const size_t        check_size =
       t8_shmem_array_get_elem_size (shmem_array);
-    SC_CHECK_ABORT (check_size == element_size,
-                    "shared memory array has wrong element size.\n");
+    EXPECT_TRUE(check_size == element_size);
```
Comment (jmark) [inline test/t8_data/t8_gtest_shmem.cxx:207]: Please transfer custom error message.

File: test/t8_data/t8_gtest_shmem.cxx [hunk 8]
```diff
@@ -132,10 +214,10 @@ t8_test_shmem_array (sc_MPI_Comm comm)
     if (t8_shmem_array_start_writing (shmem_array)) {
       /* Double check that array_length is big enough so that each of the three cases
        * is covered. */
-      SC_CHECK_ABORT (0 < array_length / 3
+      EXPECT_TRUE(0 < array_length / 3
```
Comment (jmark) [inline test/t8_data/t8_gtest_shmem.cxx:217]: Please transfer custom error message.

File: test/t8_data/t8_gtest_shmem.cxx [hunk 9]
```diff
@@ -152,20 +234,18 @@ t8_test_shmem_array (sc_MPI_Comm comm)
     }
     t8_shmem_array_end_writing (shmem_array);
 
+    /* Check value at each position */
     for (int i = 0; i < array_length; ++i) {
       t8_gloidx_t         value = t8_shmem_array_get_gloidx (shmem_array, i);
-      SC_CHECK_ABORTF (value == i,
-                       "Value at position %i not correct (expected %i, got %li)\n",
-                       i, i, value);
+      EXPECT_TRUE(value == i);
```
Comment (jmark) [inline test/t8_data/t8_gtest_shmem.cxx:240]: Please transfer custom error message.

File: test/t8_data/t8_gtest_shmem.cxx [hunk 10]
```diff
@@ -152,20 +234,18 @@ t8_test_shmem_array (sc_MPI_Comm comm)
...
 
     /* Copy */
     t8_shmem_array_t    copy_array;
     t8_shmem_array_init (&copy_array, element_size, array_length, comm);
-    t8_debugf ("Copying array.\n");
     t8_shmem_array_copy (copy_array, shmem_array);
-    SC_CHECK_ABORT (t8_shmem_array_is_equal (copy_array, shmem_array),
-                    "Array are not equal after copy.");
+    /* Check equality of arrays after copying. */
+    EXPECT_TRUE(t8_shmem_array_is_equal (copy_array, shmem_array));
```
Comment (jmark) [inline test/t8_data/t8_gtest_shmem.cxx:248]: Please transfer custom error message.

**Review by jmark — CHANGES_REQUESTED** [review]
This section shows you how to add custom error messages to you assertions:
https://google.github.io/googletest/primer.html#assertions

**Comment by WeberJulia** [comment]
compare PR #421

**Pull request closed without merging**